crates/storage/provider/src/providers/snapshot/manager.rs

```diff
@@ -223,39 +219,90 @@ impl SnapshotProvider {
             }
             let tx_start = snapshots_rev_iter.peek().map(|(tx_end, _)| *tx_end + 1).unwrap_or(0);
             if tx_start <= tx {
-                return Some((block_range.clone(), tx_start..=*tx_end))
+                return Some(find_fixed_range(BLOCKS_PER_SNAPSHOT, *block_range.end()))
             }
         }
         None
     }
 
-    /// Updates the inner transaction and block index
-    pub fn update_index(&self) -> ProviderResult<()> {
-        let mut block_index = self.snapshots_block_index.write();
+    /// Updates the inner transaction and block index alongside the internal `self.map`.
+    ///
+    /// Any entry higher than `segment_max_block` will be deleted from the previous structures.
+    pub fn update_index(
+        &self,
+        segment: SnapshotSegment,
+        segment_max_block: Option<BlockNumber>,
```

Comment:
when it can be `None`? if no snapshot exists for this segment? Let's mention this in the comment